x/erc20/keeper/msg_server.go

```diff
@@ -94,197 +95,7 @@ func (k Keeper) ConvertERC20(
-// token pair:
-//   - escrow coins on module account
-//   - mint tokens and send to receiver
-//   - check if token balance increased by amount
-func (k Keeper) convertCoinNativeCoin(
-	ctx sdk.Context,
-	pair types.TokenPair,
-	msg *types.MsgConvertCoin,
-	receiver common.Address,
-	sender sdk.AccAddress,
-) (*types.MsgConvertCoinResponse, error) {
-	// NOTE: ignore validation from NewCoin constructor
-	coins := sdk.Coins{msg.Coin}
-	erc20 := contracts.ERC20MinterBurnerDecimalsContract.ABI
-	contract := pair.GetERC20Contract()
-	balanceToken := k.BalanceOf(ctx, erc20, contract, receiver)
-	if balanceToken == nil {
-		return nil, errorsmod.Wrap(types.ErrEVMCall, "failed to retrieve balance")
-	}
-
-	// Escrow coins on module account
-	err := k.bankKeeper.SendCoinsFromAccountToModule(ctx, sender, types.ModuleName, coins)
-	if err != nil {
-		return nil, errorsmod.Wrap(err, "failed to escrow coins")
-	}
-
-	// Mint tokens and send to receiver
-	_, err = k.CallEVM(ctx, erc20, types.ModuleAddress, contract, true, "mint", receiver, msg.Coin.Amount.BigInt())
-	if err != nil {
-		return nil, err
-	}
-
-	// Check expected receiver balance after transfer
-	tokens := msg.Coin.Amount.BigInt()
-	balanceTokenAfter := k.BalanceOf(ctx, erc20, contract, receiver)
-	if balanceTokenAfter == nil {
-		return nil, errorsmod.Wrap(types.ErrEVMCall, "failed to retrieve balance")
-	}
-	expToken := big.NewInt(0).Add(balanceToken, tokens)
-
-	if r := balanceTokenAfter.Cmp(expToken); r != 0 {
-		return nil, errorsmod.Wrapf(
-			types.ErrBalanceInvariance,
-			"invalid token balance - expected: %v, actual: %v", expToken, balanceTokenAfter,
-		)
-	}
-
-	defer func() {
-		telemetry.IncrCounterWithLabels(
-			[]string{"tx", "msg", "convert", "coin", "total"},
-			1,
-			[]metrics.Label{
-				telemetry.NewLabel("denom", pair.Denom),
-			},
-		)
-
-		if msg.Coin.Amount.IsInt64() {
-			telemetry.IncrCounterWithLabels(
-				[]string{"tx", "msg", "convert", "coin", "amount", "total"},
-				float32(msg.Coin.Amount.Int64()),
-				[]metrics.Label{
-					telemetry.NewLabel("denom", pair.Denom),
-				},
-			)
-		}
-	}()
-
-	ctx.EventManager().EmitEvents(
-		sdk.Events{
-			sdk.NewEvent(
-				types.EventTypeConvertCoin,
-				sdk.NewAttribute(sdk.AttributeKeySender, msg.Sender),
-				sdk.NewAttribute(types.AttributeKeyReceiver, msg.Receiver),
-				sdk.NewAttribute(sdk.AttributeKeyAmount, msg.Coin.Amount.String()),
-				sdk.NewAttribute(types.AttributeKeyCosmosCoin, msg.Coin.Denom),
-				sdk.NewAttribute(types.AttributeKeyERC20Token, pair.Erc20Address),
-			),
-		},
-	)
-
-	return &types.MsgConvertCoinResponse{}, nil
-}
-
-// convertERC20NativeCoin handles the erc20 conversion for a native Cosmos coin
-// token pair:
-//   - burn escrowed tokens
-//   - unescrow coins that have been previously escrowed with ConvertCoin
-//   - check if coin balance increased by amount
-//   - check if token balance decreased by amount
-func (k Keeper) convertERC20NativeCoin(
```

Comment:
@Vvaradinov this function must be used for the conversion in the upgrade handler.

We should make it public and add a `Deprecate` comment